internal/fields/mappings.go

```diff
@@ -0,0 +1,691 @@
+		return errs.Unique()
+	}
+
+	mappingErrs := v.compareMappings("", rawPreview, rawActual)
+	errs = append(errs, mappingErrs...)
+
+	if len(errs) > 0 {
+		return errs.Unique()
+	}
+
+	return nil
+}
+
+func currentMappingPath(path, key string) string {
+	if path == "" {
+		return key
+	}
+	return fmt.Sprintf("%s.%s", path, key)
+}
+
+func mappingParameter(field string, definition map[string]any) string {
+	fieldValue, ok := definition[field]
+	if !ok {
+		return ""
+	}
+	value, ok := fieldValue.(string)
+	if !ok {
+		return ""
+	}
+	return value
+}
+
+func isLocalFieldTypeArray(field string, schema []FieldDefinition) bool {
+	definition := findElementDefinitionForRoot("", field, schema)
+	if definition == nil {
+		return false
+	}
+	if definition.External != "" {
+		return false
+	}
+	return definition.Type == "array"
+}
+
+func isEmptyObject(definition map[string]any) bool {
+	// Example:
+	//  "_tmp": {
+	//    "type": "object"
+	//  },
+	if len(definition) != 1 {
+		return false
+	}
+	return mappingParameter("type", definition) == "object"
+}
+
+func isObject(definition map[string]any) bool {
+	// Example:
+	// "http": {
+	//   "properties": {
+	// 	   "request": {
+	// 	     "properties": {
+	//         "method": {
+	//           "type": "keyword",
+	//           "ignore_above": 1024
+	//         }
+	//       }
+	//     }
+	//   }
+	// }
+	field, ok := definition["properties"]
+	if !ok {
+		return false
+	}
+	if _, ok = field.(map[string]any); !ok {
+		return false
+	}
+	return true
+}
+
+func isObjectFullyDynamic(definition map[string]any) bool {
+	// Example:
+	//  "labels": {
+	//    "type": "object",
+	//    "dynamic": "true"
+	//  },
+	fieldType := mappingParameter("type", definition)
+	fieldDynamic := mappingParameter("dynamic", definition)
+
+	if fieldType != "object" {
+		return false
+	}
+	if fieldDynamic != "true" {
+		return false
+	}
+
+	field, ok := definition["properties"]
+	if !ok {
+		return true
+	}
+	props, ok := field.(map[string]any)
+	if !ok {
+		return false
+	}
+	// It should not have properties
+	// https://www.elastic.co/guide/en/elasticsearch/reference/8.16/dynamic.html
+	if len(props) != 0 {
+		return false
+	}
+	return true
+}
+
+func isMultiFields(definition map[string]any) bool {
+	// Example:
+	//  "path": {
+	//    "type": "keyword",
+	//    "fields": {
+	//      "text": {
+	//        "type": "match_only_text"
+	//      }
+	//    }
+	//  },
+	fieldType := mappingParameter("type", definition)
+	if fieldType == "" {
+		return false
+	}
+	field, ok := definition["fields"]
+	if !ok {
+		return false
+	}
+	if _, ok = field.(map[string]any); !ok {
+		return false
+	}
+	return true
+}
+
+func isNumberTypeField(previewType, actualType string) bool {
+	if slices.Contains([]string{"float", "long", "double"}, previewType) && slices.Contains([]string{"float", "long", "double"}, string(actualType)) {
+		return true
+	}
+
+	return false
+}
```

Comment:
Added this method to compare different types for numbers. This tries to mimic this logic: https://github.com/elastic/elastic-package/blob/8cc126ae5015dd336b22901c365e8c98db4e7c15/internal/fields/validate.go#L1234-L1247

Would it be to broad this comparison?

Reply:
Currently, it looks like there is just one package (`ibmmq`) failing with this error when using mappings to validate. That package is not failing with the current approach of validating fields.